src/validation.h

```diff
@@ -894,7 +921,7 @@ class ChainstateManager
     void Reset();
 };
 
-extern ChainstateManager g_chainman;
+extern ChainstateManager g_chainman GUARDED_BY(::cs_main);
```

Comment:
abfc152

If understand well the new lock model, `g_chainman` content is under `cs_main` but not `CChainState`. If yes, why some `GetAll` access in init.cpp don't use lock for access ?

Reply:
re: https://github.com/bitcoin/bitcoin/pull/17737#discussion_r385962192

> If understand well the new lock model, `g_chainman` content is under `cs_main` but not `CChainState`. If yes, why some `GetAll` access in init.cpp don't use lock for access ?

I don't think this is true. Maybe it was resolved? You can add EXCLUSIVE_LOCKS_REQUIRED(cs_main) to the GetAll declaration, and it will only fail in test code using a local chain state manager not tied to the cs_main global